api/v3/Case.php

```diff
@@ -240,6 +240,22 @@ function civicrm_api3_case_get($params) {
     $sql->where("a.id IN (SELECT case_id FROM civicrm_case_contact WHERE $clause)");
   }
 
+  // Order by case contact (primary client)
+  if (!empty($options['sort']) && strpos($options['sort'], 'contact_id') === 0) {
+    $sort = explode(', ', $options['sort']);
+    $contactSort = array_shift($sort);
+    $params['options']['sort'] = implode(', ', $sort);
+    unset($params['option_sort'], $params['option.sort'], $params['sort']);
+    list($sortField, $dir) = array_pad(explode(' ', $contactSort), 2, 'ASC');
+    list(, $sortField) = array_pad(explode('.', $sortField), 2, 'id');
```

Comment:
Repeating aloud for clarity: The `$sortField` could contain sketchy characters, but the check `!array_key_exists($sortField,...` protects against that.